packages/ai/src/ai/common/models/vision/vision.py

```
@@ -0,0 +1,428 @@
+            results.append(extracted)
+
+        return results
+
+    @staticmethod
+    def _get_memory_footprint(model: Any) -> float:
+        """Get actual GPU memory footprint."""
+        try:
+            total_params = sum(p.numel() for p in model.parameters())
+            total_bytes = total_params * 4 * 1.3
+            return round(total_bytes / 1073741824, 2)
+        except Exception:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)